### PR TITLE
Update stats in process

### PR DESCRIPTION
**Pull request by russelldb**
When a node is giving good thoughput then riak_kv_stat cannot keep
up. Large message queues ensue, memory balloons, the node dies.
This commit only forces stat creation through the gen_server, any
other stat update is direct in process.

This does cause a degradation in throughput & latency as each get / put
process is waiting to acquire the locks on the folsom tables.

NOTE: this works a whole lot better with these `folsom` optimisations https://github.com/basho/folsom/tree/adt-faster-spiral

With the `folsom` improvements, put throughput is as good as 1.2.1 (though worse than master without stats.) Even with the `folsom` improvements, there is extra latency, most noticeable in the 95th%ile.

### CODE REVIEW

**Comment by joedevivo** [comment]
./riak_test -t verify_riak_stats -> pass

**Comment by engelsanchez** [comment]
The change is simple looks good as far as I can tell. :+1: Was that your new plus one @joedevivo ?

File: src/riak_kv_stat.erl [hunk 1]
```diff
@@ -85,8 +99,9 @@ init([]) ->
...
     {ok, ok}.
 
-handle_call(_Req, _From, State) ->
-    {reply, ok, State}.
+handle_call({register, Name, Type}, _From, State) ->
+    Rep = do_register_stat(Name, Type),
+    {reply, Rep, State}.
 
 handle_cast({update, Arg}, State) ->
     do_update(Arg),
```
Comment (joedevivo) [inline src/riak_kv_stat.erl]: You should remove this handle_cast clause, since it's no longer used.

**Comment by joedevivo** [comment]
@engelsanchez no, that was just saying the code passed the tests, still was awaiting manual review. I told @russelldb  I'd remove the extra handle_cast and run riak_test again.

**Comment by joedevivo** [comment]
Ok, riak_test passed again after the clause was removed. +1

**Pull request merged**